src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -329,13 +335,23 @@ def _do_upgrade(self):
                     return
                 logger.info('Upgrade: Redeploying %s.%s' %
                             (d.daemon_type, d.daemon_id))
-                self.mgr._daemon_action(
-                    d.daemon_type,
-                    d.daemon_id,
-                    d.hostname,
-                    'redeploy',
-                    image=target_image
-                )
+                try:
+                    self.mgr._daemon_action(
+                        d.daemon_type,
+                        d.daemon_id,
+                        d.hostname,
+                        'redeploy',
+                        image=target_image
+                    )
+                except Exception as e:
+                    self._fail_upgrade('UPGRADE_REDEPLOY_DAEMON', {
+                        'severity': 'warning',
+                        'summary': f'Upgrade daemon: {d.name()}: {e}',
+                        'count': 1,
+                        'detail': [
+                            f'Upgrading daemon {d.name()} on host {d.hostname} failed.',
+                        ],
```

Comment:
```suggestion
                        'summary': f'Upgrading daemon {d.name()} on host {d.hostname} failed.',
                        'count': 1,
                        'detail': [
                            f'Upgrade daemon: {d.name()}: {e}'
                        ],
```
nit, but swap the summary and detail messages? seems like the exceptions cause should be part of the detail ..